### PR TITLE
Feat/1362 fzcurrency minmax

### PR DESCRIPTION
**Pull request by Trebh**
Jira: [LIB-1362](https://fiscozen.atlassian.net/browse/LIB-1362)

[LIB-1362]: https://fiscozen.atlassian.net/browse/LIB-1362?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/composables/src/composables/useCurrency.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import { Ref, watch, getCurrentInstance, computed, ref, nextTick } from 'vue'
 import { FzUseCurrencyOptions } from '../types'
-import { format as formatNumber, parse } from '../utils'
+import { format as formatNumber, parse, roundTo } from '../utils'
 
 export const useCurrency = (options: FzUseCurrencyOptions) => {
   const inputRef: Ref<HTMLInputElement | null | undefined> = ref(null)
   const vm = getCurrentInstance()
```
Comment (Crish15) [inline packages/composables/src/composables/useCurrency.ts:7]: Accessing the current instance has made this composable highly coupled with the `FzCurrencyInput`: components using this composable must have an `amount` and a `nullOnEmpty` prop, as well as an emitter for the `update:amount` event.  

For now, this is not a major issue, but we should probably create a refactor card to move these composables into the input package.

File: packages/input/src/FzCurrencyInput.vue [hunk 2]
```diff
@@ -107,10 +118,21 @@ onMounted(() => {
...
   });
 });
-const model = defineModel("amount");
+const model = defineModel<number>("amount");
+
+const stepUpDown = (amount: number) => {
+  if (!props.step) {
+    return;
+  }
+  let stepVal = props.forceStep ? roundTo(props.step, model.value || 0) : model.value || 0
```
Comment (Crish15) [inline packages/input/src/FzCurrencyInput.vue:127]:
```suggestion
  const value = model.value || 0;
  let stepVal = props.forceStep ? roundTo(props.step, value) : value
```

File: packages/composables/src/composables/useCurrency.ts (hunk 1 above)
Comment (Trebh) [reply packages/composables/src/composables/useCurrency.ts:7]: Correct. The reason I made it this way is I loosely followed what they did in the existing components we used as examples. 
I kind of understnad why it was done this way, it's quite hard to decouple it from components while maintaining reactivity

Comment (Trebh) [reply packages/composables/src/composables/useCurrency.ts:7]: although I would argue this specific PR has little to do with it. If anything, I added utils (which might be a better solution, along with a refactor that moves component logic... well into the component itself I guess)

**Pull request merged**